ribs/archives/_cvt_archive.py

```diff
@@ -260,12 +264,30 @@ def index_of(self, measures_batch):
             _, indices = self._centroid_kd_tree.query(measures_batch)
             return indices.astype(np.int32)
 
+        expanded_dims = np.expand_dims(measures_batch, axis=1)
+        if self.chunk_distances:  # Compute indices chunks at a time
+            distances = np.sum(np.square(
+                expanded_dims[0:min(len(measures_batch), self.chunk_size)] -
+                self.centroids),
+                               axis=2)
+            indices = np.argmin(distances, axis=1).astype(np.int32)
+            for i in range(
+                    2,
+                    len(measures_batch) // self.chunk_size +
+                    len(measures_batch) % self.chunk_size + 1):
+                l = (i - 1) * self.chunk_size
+                r = min(len(measures_batch), i * self.chunk_size)
```

Comment:
You don't actually need this `min` in this implementation because numpy handles OOB errors gracefully. For example, the below code outputs `[1,2,3,4]`
```python
a = np.array([0,1,2,3,4])
print(a[1:100])
```

Reply:
Just keep the bounds thing -- it won't affect runtime